### PR TITLE
add update_nft_class_credit brun_nft

### PR DESCRIPTION
**Pull request by baidang201**

### CODE REVIEW

File: pallets/credit-accumulation/Cargo.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ node-primitives = {version = '2.0.0', default-features = false, path = "../../pr
 pallet-balances = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.17", default-features = false}
 pallet-credit = { default-features = false, path = "../credit", version = "3.0.0" }
 pallet-deeper-node = { version = "3.0.0", default-features = false, path = "../deeper-node" }
+pallet-uniques = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.17", default-features = false}
```
Comment (xcaptain) [inline pallets/credit-accumulation/Cargo.toml:26]: I found  that `pallet-uniques` was added to `credit-accumulation`, `micropayment`, `staking`, `tips` and I think these crates don't need to change.
You are using tightly coupling, maybe using loosely coupling will solve this problem
https://docs.substrate.io/v3/runtime/pallet-coupling/

File: pallets/tips/src/tests.rs [hunk 2]
```diff
@@ -157,6 +159,36 @@ impl pallet_treasury::Config for Test {
...
+    pub const ValueLimit: u32 = 256;
+    pub const StringLimit: u32 = 50;
+}
+
+impl pallet_uniques::Config for Test {
+    type Event = Event;
+    type ClassId = u32;
+    type InstanceId = u32;
+    type Currency = Balances;
+    type ForceOrigin = frame_system::EnsureRoot<u128>;
```
Comment (xcaptain) [inline pallets/tips/src/tests.rs:180]: why using a `u128`? we are using `EnsureRoot<AccountId>` elsewhere.
also `EnsureRoot` was already imported, no need to use its full path.

File: pallets/tips/src/tests.rs [hunk 3]
```diff
@@ -157,6 +159,36 @@ impl pallet_treasury::Config for Test {
...
+    type ClassDeposit = ConstU64<2>;
```
Comment (xcaptain) [inline pallets/tips/src/tests.rs:181]: These 4 `deposit` values are of type `BalanceOf`, maybe you should set a `1*DPR` or `2*DPR`, just write a 1 or 2 is a very small value.

File: pallets/tips/src/tests.rs [hunk 4]
```diff
@@ -157,6 +159,36 @@ impl pallet_treasury::Config for Test {
...
+    type InstanceDeposit = ConstU64<1>;
+    type MetadataDepositBase = ConstU64<1>;
+    type AttributeDepositBase = ConstU64<1>;
+    type DepositPerByte = ConstU64<1>;
+    type StringLimit = ConstU32<50>;
+    type KeyLimit = ConstU32<50>;
```
Comment (xcaptain) [inline pallets/tips/src/tests.rs:187]: These 3 `limit` values were set to 50, is it enough? can we change it in the future?

File: pallets/tips/src/tests.rs (hunk 3 above)
Comment (baidang201) [reply pallets/tips/src/tests.rs:181]: the test config is small value, but big value in runtime/src/lib.rs
```
    pub const ClassDeposit: Balance = 100 * DOLLARS;
    pub const InstanceDeposit: Balance = 1 * DOLLARS;
    
    pub const MetadataDepositBase: Balance = 10 * DPR;
    pub const MetadataDepositPerByte: Balance = 1 * DPR;
```

File: pallets/tips/src/tests.rs (hunk 4 above)
Comment (baidang201) [reply pallets/tips/src/tests.rs:187]: in runtime/src/lib.rs
```
    pub const KeyLimit: u32 = 32;
    pub const ValueLimit: u32 = 256;
    pub const StringLimit: u32 = 50;
```
    KeyLimit and ValueLimit would be enough, StringLimit change to 256 would be fine.

File: pallets/tips/src/tests.rs (hunk 2 above)
Comment (baidang201) [reply pallets/tips/src/tests.rs:180]: i try remove "frame_system::", but EnsureRoot have no imported.

1 there are not alias like "type AccountId = u64;"
2 "type AccountId = u128; // u64 is not enough to hold bytes used to generate bounty account" in  "impl frame_system::Config for Test"

**Review by AmadeusGB — APPROVED** [review]

**Pull request merged**